src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs

```diff
@@ -49,30 +49,45 @@ internal static string GetPJorPRCacheFilePath(RestoreRequest request)
         internal static string GetToolCacheFilePath(RestoreRequest request, LockFile lockFile)
         {
 
-            if (request.ProjectStyle != ProjectStyle.DotnetCliTool)
+            if (request.ProjectStyle == ProjectStyle.DotnetCliTool && lockFile != null)
             {
                 var toolName = ToolRestoreUtility.GetToolIdOrNullFromSpec(request.Project);
                 var lockFileLibrary = ToolRestoreUtility.GetToolTargetLibrary(lockFile, toolName);
 
                 if (lockFileLibrary != null)
                 {
                     var version = lockFileLibrary.Version;
-
                     var toolPathResolver = new ToolPathResolver(request.PackagesDirectory);
-                    var projFileName = Path.GetFileName(request.Project.RestoreMetadata.ProjectPath);
-                    return PathUtility.GetDirectoryName(toolPathResolver.GetLockFilePath(
+
+                    return Path.Combine(toolPathResolver.GetToolDirectoryPath(
                         toolName,
                         version,
-                        lockFile.Targets.First().TargetFramework)) + $"{projFileName}.nuget.cache";
+                        lockFile.Targets.First().TargetFramework), $"{toolName}.nuget.cache");
                 }
             }
             return null;
         }
+
+        internal static string GetToolCacheFilePath(string toolDirectory, string toolName)
+        {
+            return Path.Combine(
+                toolDirectory,
+                 $"{toolName}.nuget.cache");
```

Comment:
lowercase the tool name, it's very possible that projects specify tool names using different casing which would cause problems on linux